src/components/Search/index.tsx

```diff
@@ -123,9 +133,41 @@ function Search({queryJSON}: SearchProps) {
         }
 
         SearchActions.search({queryJSON, offset});
-        // eslint-disable-next-line react-compiler/react-compiler, react-hooks/exhaustive-deps
     }, [isOffline, offset, queryJSON]);
 
+    const searchTriggeredRef = useRef(false);
+
+    useEffect(() => {
+        const previousTransactionsLength = previousTransactions && Object.keys(previousTransactions).length;
+        const transactionsLength = transactions && Object.keys(transactions).length;
+
+        if (!previousTransactionsLength || !transactionsLength || previousTransactionsLength === transactionsLength) {
+            return;
+        }
+
+        // Check if the search block was already triggered
+        if (!searchTriggeredRef.current && transactionsLength > previousTransactionsLength) {
+            // A transaction was added, trigger the action again
+            SearchActions.search({queryJSON, offset});
+            searchTriggeredRef.current = true; // Set the flag to true to prevent further triggers
+        }
+    }, [transactions, previousTransactions, queryJSON, offset]);
+
+    // Reset the flag using InteractionManager after the search action
+    useEffect(() => {
+        if (!searchTriggeredRef.current) {
+            return;
+        }
+
+        // Schedule a task to reset the flag after all current interactions have completed
+        const interactionHandle = InteractionManager.runAfterInteractions(() => {
+            searchTriggeredRef.current = false; // Reset the flag after interactions
+        });
+
```

Comment:
@luacmartins Ideally, we would update the `Search` snapshot in BE and pop the newly added transaction in the search list via pusher and we wouldn't have to check transactions / previous transactions in this way - but it's better than performing a full page reload from a UX point of view, given that we want to pop the row in with a nice animation.